### ISSUE TITLE
Remove the ability to block until speech is complete (wait=True)

### ISSUE THREAD

**Issue by nvdaTrac** [nvaccess/test_nvdaTracImport#197]
**Reported by jteh on 19 Jun 2008 02:01 UTC**
NVDA provides the ability to block (i.e. nothing else will be processed) until speech is complete. This is done by passing wait=True to the functions which speak text, including speech.speakText, speech.speakMessage and the speakText function in synth drivers. This functionality is severely broken for every synth driver except SAPI5. For utterances longer than two or three words, it speaks the text but then NVDA freezes. Even if this could be fixed, the functionality is flawed, as it means the user cannot silence speech before the text is spoken. It is usually used to prevent future events from silencing the utterance, but a better solution should be found which does not prevent the user from silencing it explicitly. Thus, this (wait=True) functionality should be removed from all functions in the speech module and all synth drivers, as well as all uses thereof.

**Comment by nvdaTrac** [comment]
**Modified by jteh on 1 Jul 2008 02:08 UTC**

**Comment by nvdaTrac** [comment]
**Comment by jteh on 1 Jul 2008 04:08 UTC**
Implemented in r2174.
